File: app/src/main/java/agi/erecreditsmanager/FreeLecture/FreeLectureLayout.java
package agi.erecreditsmanager.FreeLecture;

import android.content.Context;
import android.view.LayoutInflater;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RelativeLayout;

import agi.erecreditsmanager.R;

public class FreeLectureLayout extends RelativeLayout {

    Context context;
    EditText nameText, creditText;
    Button addButton;

    public FreeLectureLayout(Context context) {
        super(context);
        this.context = context;

        init();
    }

    private void init() {
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        inflater.inflate(R.layout.freelecture, this, true);

        nameText = (EditText) findViewById(R.id.nameText);
        creditText = (EditText) findViewById(R.id.creditText);
        addButton = (Button) findViewById(R.id.addButton);
    }

    public Button getAddButton() {
        return addButton;
    }

    public String getFreeName() {
        return nameText.getText().toString();
    }

    public int getFreeCredit() {
        return Integer.parseInt(creditText.getText().toString());
    }

    public void setInitiation() {
        nameText.setHint("      과목명 입력");
        creditText.setHint("학점 입력");
    }
}

File: app/src/main/java/agi/erecreditsmanager/FreeLecture/FreeLectureLayout.kt
package agi.erecreditsmanager.FreeLecture

import agi.erecreditsmanager.R
import android.content.Context
import android.view.LayoutInflater
import android.widget.Button
import android.widget.RelativeLayout
import kotlinx.android.synthetic.main.freelecture.view.*

class FreeLectureLayout(context : Context) : RelativeLayout(context) {
    init {
        LayoutInflater.from(context).run {
            inflate(R.layout.freelecture, this@FreeLectureLayout, true)

            nameText
            creditText
            addButton
            editButton
        }
    }

    fun getAddButton() : Button {
        return addButton
    }
    fun getEditButton() : Button {
        return editButton
    }

    fun getFreeName() : String {
        return nameText.text.toString()
    }

    fun getFreeCredit() : Int {
        return Integer.parseInt(creditText.text.toString())
    }

    fun setInitiation() {
        nameText.hint = "      과목명 입력"
        creditText.hint = "학점 입력"
    }
}